lms/djangoapps/courseware/courses.py

```diff
@@ -32,6 +32,21 @@
 from student.models import CourseEnrollment
 import branding
 
+# Libraries used for the updates filtered by date and locale date feature. 
+from xml.dom import minidom
+import lxml
+from lxml import etree
+from dateutil import parser
+from django.utils import timezone
+import datetime
+from django.utils.translation import ugettext as _, get_language
+import locale
+from openedx.core.djangoapps.user_api.models import UserPreference
+from lang_pref import LANGUAGE_KEY
+from django.utils.dateformat import format
+from django.utils import formats
```

Comment:
These imports need to be reorganized. See https://github.com/edx/edx-platform/wiki/Python-Guidelines#imports-order